controllers/storagecluster/storageclasses.go

```diff
@@ -103,6 +104,27 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 				skippedSC = append(skippedSC, sc.Name)
 				continue
 			}
+		case strings.Contains(sc.Name, "-ceph-replica1-rbd") && !scc.isClusterExternal:
+			// wait for CephBlockPools to be ready
+			cephBlockPools := cephv1.CephBlockPoolList{}
+			err := r.Client.List(context.TODO(), &cephBlockPools, client.InNamespace(sc.Parameters["clusterID"]))
+			num := len(initData.Status.FailureDomainValues)
+			// if the number of CephBlockPools is less than the number of failure domains+1, skip reconciling StorageClass
+			if err != nil || len(cephBlockPools.Items) < num+1 {
+				skippedSC = append(skippedSC, sc.Name)
+				continue
+			}
```

Comment:
This shouldn't be needed at all, because the total number of non-resilient pools should not depend on the number of resilient pools. What I would do is start with `num := 0`, then increment it when you loop over the block pool list and find a "Ready" non-resilient pool. After that is when you should check if the correct number of pools are ready.